public/app/features/serviceaccounts/ServiceAccountsListPage.tsx

```diff
@@ -33,12 +34,46 @@ const mapDispatchToProps = {
 
 const connector = connect(mapStateToProps, mapDispatchToProps);
 
-const ServiceAccountsListPage: React.FC<Props> = ({ loadServiceAccounts, navModel, serviceAccounts, isLoading }) => {
+const ServiceAccountsListPage: React.FC<Props> = ({
+  loadServiceAccounts,
+  updateServiceAccount,
+  navModel,
+  serviceAccounts,
+  isLoading,
+}) => {
   const styles = useStyles2(getStyles);
+  const [roleOptions, setRoleOptions] = useState<Role[]>([]);
+  const [builtinRoles, setBuiltinRoles] = useState<Record<string, Role[]>>({});
 
   useEffect(() => {
+    async function fetchOptions() {
```

Comment:
I think the same logic is used in other places where role picker is used, I think it'd make sense to extract it into a custom hook.

Reply:
Yes, this one particularly comes from team list

Reply:
It doesn't really map to exactly the same usage as in other places (and looking at it I'm pretty sure it's bound to change soon) but I've moved this logic to redux